### PR TITLE
[CDAP-17744] Fixes Schema editor to surface UI validation errors.

### PR DESCRIPTION
**Pull request by ajainarayanan**
JIRA - https://cdap.atlassian.net/browse/CDAP-17744

### CODE REVIEW

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaValidator/index.tsx [hunk 1]
```diff
@@ -66,13 +67,13 @@ class SchemaValidatorProvider extends React.Component<ISchemaValidatorProviderPr
   };
 
   public componentWillReceiveProps(nextProps) {
-    const { errors } = nextProps;
-    if (errors === null) {
+    const { errors, reset } = nextProps;
+    if (reset) {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaValidator/index.tsx:71]: might not be a concern and also not sure what should the alternative be either, but is there a potential for a race condition here? where the props for the errors map gets updated first, and then the reset props set to false in a different digest cycle? that could potentially erase the errorMap

**Review by elfenheart — APPROVED** [review]
Just have a single question that might not be a concern. Otherwise LGTM

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaValidator/index.tsx (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaValidator/index.tsx:71]: Hhmm from the angular side we reset the validating flag and at the same time populate the error map.. React will batch state updates which will then trigger a prop update. So SchemaValidator componentWillReceiveProps will be called only after the setState in schemaeditor is done which will be batched even if the properties update async. This is what I noticed while testing as well. Let me know if I am missing anything.

**Comment by ajainarayanan** [comment]
Schema e2e tests successfully ran locally.
Screen Shot 2021-03-04 at 9 20 21 PM

**Pull request merged**